### PR TITLE
ARROW-1591: C++: Xcode 9 is not correctly detected

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: cpp/cmake_modules/CompilerInfo.cmake [hunk 1]
```diff
@@ -68,6 +68,11 @@ elseif("${COMPILER_VERSION_FULL}" MATCHES ".*clang-8")
   set(COMPILER_FAMILY "clang")
   set(COMPILER_VERSION "3.8.0svn")
 
+# clang on Mac OS X, XCode 8.
+elseif("${COMPILER_VERSION_FULL}" MATCHES ".*clang-9")
+  set(COMPILER_FAMILY "clang")
+  set(COMPILER_VERSION "4.0.0svn")
```
Comment (xhochy) [inline cpp/cmake_modules/CompilerInfo.cmake]: This is just guessing, I have not found any public source that clarifies this. At the moment, we're not using the version, only the family, so this should not matter that much.

**Comment by wesm** [comment]
Seems like something went wrong when installing the Sphinx docs requirements, causing a NumPy version conflict:

https://travis-ci.org/apache/arrow/jobs/278165398#L9068

 Unrelated to this change

**Comment by xhochy** [comment]
@wesm any idea on how to fix this?

**Comment by wesm** [comment]
It is probably related to a package update in conda-forge causing an issue with the package dependency graph. I will take a look

File: cpp/cmake_modules/CompilerInfo.cmake (hunk 1 above)
Comment (wesm) [reply cpp/cmake_modules/CompilerInfo.cmake]: OK, we can always revisit if this comes up

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**